packages/eslint-plugin/docs/rules/no-unnecessary-condition.mdx

```diff
@@ -107,7 +107,7 @@ function assert(condition: unknown): asserts condition {
 
 assert(false); // Unnecessary; condition is always falsy.
 
-const neverNull = {};
+const neverNull = { someProperty: 'someValue' };
```

Comment:
> That's due to TS's (questionable?) decision to give `neverNull` type `{}` instead of type `object`, even when initializing a `const` variable.

😩 this again... is there a TypeScript issue filed that we can reference? It feels like a bug to me.

I also don't mind it very much. It's not often that folks use `{}`. This docs change kind of makes it more realistic to me.

Reply:
> 😩 this again... is there a TypeScript issue filed that we can reference? It feels like a bug to me.

That's a good question. I'll take an action item to investigate this.

> I also don't mind it very much. It's not often that folks use `{}`. This docs change kind of makes it more realistic to me.

Agreed that the docs change is probably an improvement. But, the false negative here makes the rule a little less understandable to me, so I'd lean on the side of wanting it to be correct if we can get it to be.

Reply:
Oh, dear, the moment I start playing around with this, I immediately find infuriating behavior around `{}`....
```ts
function takesObject(x: object) {
  if (x) {
    console.log('should always be true')
  } else {
    console.error('this should be unreachable');
  }
}
const o: {} = 0;
// no error here!?!?!?!?
takesObject(o);
```

Reply:
Regarding `const o = {}; // has type {}`, started discord conversation at https://discord.com/channels/508357248330760243/508357638677856287/1310419206260654081.

Regarding `{}` -> `object` assignments, filed https://github.com/microsoft/TypeScript/issues/60582... and immediately learned this is a duplicate of https://github.com/microsoft/TypeScript/issues/56205, in which this behavior is deemed intentional 🫤.

Reminder to self: I'm pretty sure we'll have some good info to add to the motivation for https://typescript-eslint.io/rules/no-empty-object-type/ as an outcome of these discussions 😆